### PR TITLE
[ci] Run scheduler unittests with GitHub Actions

### LINKED ISSUE #3261: Use `docker compose` to run the slurm related unit tests
The idea is to use a similar setup to the tutorial for [multi-node tests](https://reframe-hpc.readthedocs.io/en/stable/tutorial.html#running-the-multi-node-examples) to simulate a slurm virtual cluster and test the slurm related functionality there.

### PR DESCRIPTION
**Pull request by teojgo**
Fixes #3261

### CODE REVIEW

File: .github/pseudo-cluster/docker-compose.yml
Comment (vkarak) [inline .github/pseudo-cluster/docker-compose.yml:1]: Are these copies of those in the tutorial examples or do they have any adaptations? Would symlinks just work for the files that are the same?

File: .github/pseudo-cluster/reframe/docker-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+cp -r /usr/local/share/reframe .
+cd reframe
+./bootstrap.sh
+
+echo "Running unittests with backend scheduler: ${BACKEND}"
+
+tempdir=$(mktemp -d -p /scratch)
+TMPDIR=$tempdir ./test_reframe.py -v \
+    --rfm-user-config=config/ci-cluster.py \
+    --rfm-user-system=pseudo-cluster:compute-${BACKEND:-squeue}
```
Comment (vkarak) [inline .github/pseudo-cluster/reframe/docker-entrypoint.sh:24]: I guess that's different from the examples.

File: config/ci-cluster.py
Comment (vkarak) [inline config/ci-cluster.py:1]: I see that we had `cscs-ci` under the `config/` dir, but maybe it's better if we put this config under `ci-scripts/configs/` along with the rest of the configs used in CI.

**Review by vkarak — CHANGES_REQUESTED** [review]
I think that we should also remove the `CSCS CI` now and the related machinery, shouldn't we?

File: .github/workflows/test-schedulers.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: Test Schedulers
```
Comment (vkarak) [inline .github/workflows/test-schedulers.yaml:1]: Just to be in sync with the rest of workflow titles.

```suggestion
name: ReFrame CI / Scheduler backend tests
```

File: .github/pseudo-cluster/docker-compose.yml
Comment (teojgo) [reply .github/pseudo-cluster/docker-compose.yml:1]: they have a small adaptation and also another adaptation in the entrypoint. I will make symlinks for the rest

File: .github/pseudo-cluster/reframe/docker-entrypoint.sh (hunk 1 above)
Comment (teojgo) [reply .github/pseudo-cluster/reframe/docker-entrypoint.sh:24]: yes this and the dockerfile which actually copies the current source code

**Review by vkarak — APPROVED** [review]

**Pull request merged**